radix-engine-interface/src/data/value_formatter.rs

```diff
@@ -256,7 +256,12 @@ pub fn format_custom_value<F: fmt::Write>(
             write!(f, "Blob(\"{}\")", value)?;
         }
         ScryptoCustomValue::NonFungibleAddress(value) => {
-            write!(f, "NonFungibleAddress(\"{}\")", value)?;
+            f.write_str("NonFungibleAddress(\"")?;
+            value
+                .resource_address()
+                .format(f, context.bech32_encoder)
+                .expect("Failed to format address");
+            write!(f, "\", {})", value.non_fungible_id())?;
```

Comment:
I'm not sure I like the fact that we rely on `display` of non fungible id being the scrypto value / manifest representation - could we have an explicit method or a method here instead?

Reply:
Updated.